src/main/java/org/mule/service/scheduler/internal/AbstractRunnableFutureDecorator.java

```diff
@@ -53,27 +61,38 @@ protected static void clearAllThreadLocals() {
   }
 
   private final int id;
-
   private volatile boolean ranAtLeastOnce = false;
   private volatile boolean started = false;
+  private ProfilingService profilingService = null;
+  private ExecutionContext initialExecutionContext = null;
 
   /**
    * @param id          a unique it for this task.
    * @param classLoader the context {@link ClassLoader} on which the {@code task} should be executed
    */
-  protected AbstractRunnableFutureDecorator(int id, ClassLoader classLoader) {
+  protected AbstractRunnableFutureDecorator(int id, ClassLoader classLoader, ProfilingService profilingService) {
     this.id = id;
     this.classLoader = classLoader;
+    // At this point, the thread is the one that scheduled the task execution. We store its context to allow its propagation.
+    if (profilingService != null) {
+      this.profilingService = profilingService;
+      initialExecutionContext = profilingService.getTracingService().getCurrentExecutionContext();
+      profilingService.getProfilingDataProducer(SCHEDULING_TASK_EXECUTION)
```

Comment:
What happens when you don't have a curent context?

Reply:
Profiling events are still emitted (will have an empty ExecutionContext). Take a look at line 133 and you will see a not null validation when propagating it.